src/__tests__/examples.test.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  buildRegex,
+  charRange,
+  choiceOf,
+  digit,
+  endOfString,
+  repeat,
+  startOfString,
+} from '../index';
+
+test('example: IPv4 address validator', () => {
+  const octet = choiceOf(
+    [digit],
+    [charRange('1', '9'), digit],
+    ['1', repeat(digit, { count: 2 })],
+    ['2', charRange('0', '4'), digit],
+    ['25', charRange('0', '5')]
+  );
+
+  const regex = buildRegex([
+    startOfString, //
```

Comment:
Unused comment?

```suggestion
    startOfString,
```

Reply:
No, inserting comment this way prevents prettier from collapsing complex expression into single line. Multiline expressions are imo more readable.